cmd/kube-scheduler/app/options/options.go

```diff
@@ -122,11 +122,11 @@ func splitHostIntPort(s string) (string, int, error) {
 	if err != nil {
 		return "", 0, err
 	}
-	portInt, err := strconv.Atoi(port)
+	portInt, err := netutils.ParsePort(port, false)
 	if err != nil {
 		return "", 0, err
 	}
-	return host, portInt, err
+	return host, uint16(port), err
```

Comment:
I don't understand this change.  You have portInt which holds the int value.  You are casting the string to uint16(), which is not the correct intention or return type.

I think you want to undo this chunk?

Reply:
Yep. I don't know why I casted to uint16 here, we already do that in the function.